### ISSUE TITLE
calendar: duplicate key on update meeting_category_rel

### ISSUE THREAD

**Issue by christophlsa** [OCA/OpenUpgrade#215]
I have the following problem:

```
2015-01-16 08:25:12,194 31261 INFO testdb_migrated OpenUpgrade: calendar: post-migration script called with version 7.0.1.0
2015-01-16 08:25:12,210 31261 ERROR testdb_migrated openerp.sql_db: bad query: update meeting_category_rel
        set event_id=calendar_event.id
        from calendar_event where event_id=openupgrade_legacy_8_0_crm_meeting_id
Traceback (most recent call last):
  File "/var/tmp/openupgrade/8.0/server/openerp/sql_db.py", line 234, in execute
    res = self._obj.execute(query, params)
IntegrityError: duplicate key value violates unique constraint "meeting_category_rel_event_id_type_id_key"
DETAIL:  Key (event_id, type_id)=(1, 1) already exists.

2015-01-16 08:25:12,215 31261 ERROR testdb_migrated OpenUpgrade: calendar: error in migration script /var/tmp/openupgrade/8.0/addons/calendar/migrations/8.0.1.0/post-migration.py: duplicate key value violates unique constraint "meeting_category_rel_event_id_type_id_key"
DETAIL:  Key (event_id, type_id)=(1, 1) already exists.

2015-01-16 08:25:12,215 31261 ERROR testdb_migrated OpenUpgrade: duplicate key value violates unique constraint "meeting_category_rel_event_id_type_id_key"
DETAIL:  Key (event_id, type_id)=(1, 1) already exists.
Traceback (most recent call last):
  File "/var/tmp/openupgrade/8.0/server/openerp/openupgrade/openupgrade.py", line 724, in wrapped_function
    func(cr, version)
  File "/var/tmp/openupgrade/8.0/addons/calendar/migrations/8.0.1.0/post-migration.py", line 179, in migrate
    import_crm_meeting(cr)
  File "/var/tmp/openupgrade/8.0/addons/calendar/migrations/8.0.1.0/post-migration.py", line 86, in import_crm_meeting
    openupgrade.get_legacy_name('crm_meeting_id'),
  File "/var/tmp/openupgrade/8.0/server/openerp/sql_db.py", line 158, in wrapper
    return f(self, *args, **kwargs)
  File "/var/tmp/openupgrade/8.0/server/openerp/sql_db.py", line 234, in execute
    res = self._obj.execute(query, params)
IntegrityError: duplicate key value violates unique constraint "meeting_category_rel_event_id_type_id_key"
DETAIL:  Key (event_id, type_id)=(1, 1) already exists.

2015-01-16 08:25:12,216 31261 ERROR testdb_migrated openerp.modules.loading: Error executing post migration script for module calendar
`-> crm
   `-> crm_profiling
: duplicate key value violates unique constraint "meeting_category_rel_event_id_type_id_key"
DETAIL:  Key (event_id, type_id)=(1, 1) already exists.

2015-01-16 08:25:12,217 31261 CRITICAL testdb_migrated openerp.service.server: Failed to initialize database `testdb_migrated`.
Traceback (most recent call last):
  File "/var/tmp/openupgrade/8.0/server/openerp/service/server.py", line 901, in preload_registries
    registry = RegistryManager.new(dbname, update_module=update_module)
  File "/var/tmp/openupgrade/8.0/server/openerp/modules/registry.py", line 324, in new
    openerp.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/var/tmp/openupgrade/8.0/server/openerp/modules/loading.py", line 398, in load_modules
    force, status, report, loaded_modules, update_module, upg_registry)
  File "/var/tmp/openupgrade/8.0/server/openerp/modules/loading.py", line 295, in load_marked_modules
    loaded, processed = load_module_graph(cr, graph, progressdict, report=report, skip_modules=loaded_modules, perform_checks=perform_checks, upg_registry=upg_registry)
  File "/var/tmp/openupgrade/8.0/server/openerp/modules/loading.py", line 215, in load_module_graph
    migrations.migrate_module(package, 'post')
  File "/var/tmp/openupgrade/8.0/server/openerp/modules/migration.py", line 179, in migrate_module
    mod.migrate(self.cr, pkg.installed_version)
  File "/var/tmp/openupgrade/8.0/server/openerp/openupgrade/openupgrade.py", line 724, in wrapped_function
    func(cr, version)
  File "/var/tmp/openupgrade/8.0/addons/calendar/migrations/8.0.1.0/post-migration.py", line 179, in migrate
    import_crm_meeting(cr)
  File "/var/tmp/openupgrade/8.0/addons/calendar/migrations/8.0.1.0/post-migration.py", line 86, in import_crm_meeting
    openupgrade.get_legacy_name('crm_meeting_id'),
  File "/var/tmp/openupgrade/8.0/server/openerp/sql_db.py", line 158, in wrapper
    return f(self, *args, **kwargs)
  File "/var/tmp/openupgrade/8.0/server/openerp/sql_db.py", line 234, in execute
    res = self._obj.execute(query, params)
IntegrityError: duplicate key value violates unique constraint "meeting_category_rel_event_id_type_id_key"
DETAIL:  Key (event_id, type_id)=(1, 1) already exists.
```

Do you had also this problem and can maybe tell me howto fast fix this?

**Comment by mayjo79** [comment]
I have a similar problem with constraint "meeting_category_rel_event_id_fkey"

I installed a new DB v7 (modules sale, crm and account) with demonstration data.
I ran migrate.py
I had 4 errors in migration.log 
2015-01-22 10:28:08,119 6246 ERROR upgrade1_migrated OpenUpgrade: Invalid value 'tree_account_reconciliation' in the table 'ir_ui_view' for the field 'type'. (1 rows).
...
2015-01-22 10:28:57,810 6246 ERROR upgrade1_migrated OpenUpgrade: Invalid value 'email' in the model 'ir_actions_server' for the field 'state'. (id 322).Please overload the new ir_actions_server._get_state function.
...
2015-01-22 10:30:40,446 6246 ERROR upgrade1_migrated OpenUpgrade: 'NoneType' object has no attribute '__getitem__'
Traceback (most recent call last):
  File "/var/tmp/openupgrade/8.0/server/openerp/openupgrade/openupgrade.py", line 716, in wrapped_function
    func(cr, version)
  File "/var/tmp/openupgrade/8.0/addons/product/migrations/8.0.1.1/post-migration.py", line 136, in migrate
    quick_request=False, binary_field=True)
  File "/var/tmp/openupgrade/8.0/server/openerp/openupgrade/openupgrade.py", line 827, in move_field_m2o
    vals = [str(x[0][:]) for x in cr.fetchall()]
TypeError: 'NoneType' object has no attribute '__getitem__'
2015-01-22 10:30:40,447 6246 ERROR upgrade1_migrated openerp.modules.loading: Error executing post migration script for module product
`-> procurement
: 'NoneType' object has no attribute '__getitem__'
2015-01-22 10:30:40,490 6246 CRITICAL upgrade1_migrated openerp.service.server: Failed to initialize database `upgrade1_migrated`.


Despite theses errors, i run Odoo 8 on this migrated database and I have the same error as @christophlsa


2015-01-22 12:53:34,701 3138 ERROR upgrade1_migrated werkzeug: Error on request:
Traceback (most recent call last):
  File "/usr/share/pyshared/werkzeug/serving.py", line 159, in run_wsgi
    execute(app)
...
  File "/opt/openerp/odoo8/odoo/openerp/sql_db.py", line 234, in execute
    res = self._obj.execute(query, params)
ParseError: "ERREUR:  une instruction insert ou update sur la table \xab meeting_category_rel \xbb viole la contrainte de cl\xe9
\xe9trang\xe8re \xab meeting_category_rel_event_id_fkey \xbb
D\xc9TAIL : La cl\xe9 (event_id)=(9) n'est pas pr\xe9sente dans la table \xab crm_meeting \xbb.
" while parsing /opt/openerp/odoo8/odoo/addons/calendar/calendar_demo.xml:20, near
<record id="calendar_event_1" model="calendar.event">
            <field eval="1" name="active"/>
            <field name="user_id" ref="base.user_root"/>
            <field name="partner_ids" eval="[(6,0,[ref('base.res_partner_6')])]"/>
            <field name="name">Follow-up for Project proposal</field>
            <field name="description">Meeting to discuss project plan and hash out the details of implementation.</field>
            <field eval="time.strftime('%Y-%m-03 10:20:00')" name="start"/>
            <field name="categ_ids" eval="[(6,0,[ref('categ_meet1')])]"/>
            <field eval="time.strftime('%Y-%m-03 16:30:00')" name="stop"/>
            <field eval="6.3" name="duration"/>
            <field eval="0" name="allday"/>
            <field name="state">open</field>
        </record>

**Comment by bramsi** [comment]
Here's what I do, some dirty tricks :smile: 

Try to remark on /var/tmp/openupgrade/8.0/addons/calendar/migrations/8.0.1.0/post-migration.py from line 79 to 94, re-run migrate.py.

Hope that help.

**Comment by pablocm-aserti** [comment]
Maybe a bit late for the OP, but today I faced the same problem with this example:
![captura de pantalla de 2016-02-12 11 21 12](https://cloud.githubusercontent.com/assets/8291685/13004997/48071806-d17f-11e5-9318-4232096945f5.png)

Details on the cause and a patch to fix it has been proposed here #435
